[4.0.11] @opentelemetry/api type error with new release

### Describe the bug

Saw this is a downstream repository:
```
node_modules/vitest/dist/chunks/traces.d.B8ukBJqA.d.ts:5:44 - error TS2307: Cannot find module '@opentelemetry/api' or its corresponding type declarations.

5 import { Context, Span, SpanOptions } from '@opentelemetry/api';
                                             ~~~~~~~~~~~~~~~~~~~~

node_modules/vitest/dist/index.d.ts:26:8 - error TS2307: Cannot find module '@opentelemetry/api' or its corresponding type declarations.

26 import '@opentelemetry/api';
          ~~~~~~~~~~~~~~~~~~~~


Found 2 errors in 2 files.

Errors  Files
     1  node_modules/vitest/dist/chunks/traces.d.B8ukBJqA.d.ts:5
     1  node_modules/vitest/dist/index.d.ts:26
```

I saw in another type-related issue you guys are suggesting to turn `skipLibCheck` on. We cannot do that. It would skip type checking **all** .d.ts files in our project (not just `node_modules`), hiding potential real issues. Already got bit by that.

### Reproduction

```sh
mkdir vitest-type-issue
cd vitest-type-issue
npm init -y
npm pkg set type=module
npm i vitest typescript
npx tsc --init --skipLibCheck false
echo "import { expect } from 'vitest'; console.log(expect);" > main.ts
npx tsc --noEmit
```

### System Info

```shell
Fedora 43
Node 24.11.1
npm 11.6.3
```

### Used Package Manager

npm

### Validations

- [x] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [x] Read the [docs](https://vitest.dev/guide/).
- [x] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vitest/src/runtime/moduleRunner/moduleEvaluator.ts
+++ b/packages/vitest/src/runtime/moduleRunner/moduleEvaluator.ts
@@ -29,6 +29,9 @@ export interface VitestModuleEvaluatorOptions {
   getCurrentTestFilepath?: () => string | undefined
   compiledFunctionArgumentsNames?: string[]
   compiledFunctionArgumentsValues?: unknown[]
+  /**
+   * @internal
+   */
   traces?: Traces
 }
 

--- a/packages/vitest/src/runtime/moduleRunner/moduleMocker.ts
+++ b/packages/vitest/src/runtime/moduleRunner/moduleMocker.ts
@@ -20,6 +20,9 @@ interface MockContext {
 
 export interface VitestMockerOptions {
   context?: vm.Context
+  /**
+   * @internal
+   */
   traces: Traces
   spyModule?: typeof import('@vitest/spy')
   root: string

--- a/packages/vitest/src/runtime/moduleRunner/moduleRunner.ts
+++ b/packages/vitest/src/runtime/moduleRunner/moduleRunner.ts
@@ -206,6 +206,9 @@ export interface VitestModuleRunnerOptions {
   getWorkerState: () => WorkerGlobalState
   mocker?: VitestMocker
   vm?: VitestVmOptions
+  /**
+   * @internal
+   */
   traces?: Traces
   spyModule?: typeof import('@vitest/spy')
   createImportMeta?: CreateImportMeta

--- a/packages/vitest/src/runtime/moduleRunner/startModuleRunner.ts
+++ b/packages/vitest/src/runtime/moduleRunner/startModuleRunner.ts
@@ -27,6 +27,9 @@ export interface ContextModuleRunnerOptions {
   context?: vm.Context
   externalModulesExecutor?: ExternalModulesExecutor
   state: WorkerGlobalState
+  /**
+   * @internal
+   */
   traces?: Traces // optional to keep backwards compat
   spyModule?: typeof import('@vitest/spy')
   createImportMeta?: CreateImportMeta

--- a/packages/vitest/src/types/worker.ts
+++ b/packages/vitest/src/types/worker.ts
@@ -49,6 +49,9 @@ export interface WorkerSetupContext {
   config: SerializedConfig
   projectName: string
   rpc: WorkerRPC
+  /**
+   * @internal
+   */
   traces: Traces
 }
 

--- a/packages/vitest/src/utils/traces.ts
+++ b/packages/vitest/src/utils/traces.ts
@@ -83,6 +83,9 @@ export class Traces {
     return !!this.#otel
   }
 
+  /**
+   * @internal
+   */
   async waitInit(): Promise<this> {
     if (this.#init) {
       await this.#init
@@ -90,6 +93,9 @@ export class Traces {
     return this
   }
 
+  /**
+   * @internal
+   */
   startContextSpan(name: string, currentContext?: Context): {
     span: Span
     context: Context
@@ -114,6 +120,9 @@ export class Traces {
     }
   }
 
+  /**
+   * @internal
+   */
   getContextFromCarrier(carrier: OTELCarrier | undefined): Context {
     if (!this.#otel) {
       return this.#noopContext
@@ -125,6 +134,9 @@ export class Traces {
     return this.#otel.propagation.extract(activeContext, carrier)
   }
 
+  /**
+   * @internal
+   */
   getContextCarrier(context?: Context): OTELCarrier | undefined {
     if (!this.#otel) {
       return undefined
@@ -177,8 +189,17 @@ export class Traces {
     }
   }
 
+  /**
+   * @internal
+   */
   $<T>(name: string, fn: (span: Span) => T): T
+  /**
+   * @internal
+   */
   $<T>(name: string, optionsOrFn: TracesSpanOptions, fn: (span: Span) => T): T
+  /**
+   * @internal
+   */
   $<T>(name: string, optionsOrFn: TracesSpanOptions | ((span: Span) => T), fn?: (span: Span) => T): T {
     const callback = typeof optionsOrFn === 'function' ? optionsOrFn : fn!
     if (!this.#otel) {
@@ -203,6 +224,9 @@ export class Traces {
     )
   }
 
+  /**
+   * @internal
+   */
   startSpan(name: string, options?: SpanOptions, context?: Context): Span {
     if (!this.#otel) {
       return this.#noopSpan
@@ -211,6 +235,9 @@ export class Traces {
     return tracer.startSpan(name, options, context)
   }
 
+  /**
+   * @internal
+   */
   async finish(): Promise<void> {
     await this.#sdk?.shutdown()
   }

